include/openssl/e_os2.h

```diff
@@ -350,6 +350,14 @@ typedef unsigned __int64 uint64_t;
 #  endif
 # endif
 
+# if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+#  define ossl_noreturn _Noreturn
+# elif defined(__GNUC__)
```

Comment:
You might need to figure out which GCC version that started supporting this attribute and adapt the condition accordingly.